x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/components/actions_menu.tsx

```diff
@@ -120,21 +130,48 @@ export const AgentPolicyActionMenu = memo<{
                   />
                 </EuiContextMenuItem>,
               ];
+
+          if (agentTamperProtectionEnabled && !agentPolicy?.is_managed) {
+            menuItems.push(
+              <EuiContextMenuItem
+                icon="minusInCircle"
+                onClick={() => {
```

Comment:
This `onClick` could be memoized.

Reply:
it's a very small callback, that may cause to re-render a very small part of the UI, so I believe it's not worth the cost of memoization, at least not until there are some performance issues

react dev tools profiler shows nothing extraordinary, too